stdlib/observability/src/main/java/org/ballerinalang/observe/nativeimpl/OpenTracerBallerinaWrapper.java

```diff
@@ -97,15 +97,21 @@ public int startSpan(String spanName, Map<String, String> tags, int parentSpanId
         if (parentSpanId == SYSTEM_TRACE_INDICATOR) {
             ObservabilityUtils.getParentContext(context).ifPresent(observerContext::setParent);
             ObservabilityUtils.setObserverContextToWorkerExecutionContext(workerExecutionContext, observerContext);
+            return attachSpan(observerContext, true, spanName);
         } else if (parentSpanId != ROOT_SPAN_INDICATOR) {
             ObserverContext parentOContext = observerContextList.get(parentSpanId);
             if (parentOContext == null) {
                 return -1;
             }
             observerContext.setParent(parentOContext);
+            return attachSpan(observerContext, true, spanName);
         }
 
-        TracingUtils.startObservation(observerContext, false);
+        return attachSpan(observerContext, false, spanName);
+    }
+    public int attachSpan(ObserverContext observerContext, boolean isClient, String spanName) {
```

Comment:
Why it's a public method? I think this method is only used within this class and hence you can declare it as private.

Reply:
Shall we also rename the method since it actually starts the span and doesn't attach to anything?